### PR TITLE
libxx:uclibxx: add patch use overload constructor of filebuf & ostream

### PR DESCRIPTION
**Pull request by zhuyanlinzyl**
.

## Summary

Bug fix in uclibxx initalize

### CODE REVIEW

File: libs/libxx/0001-uClibcxx-ios-fix-wclog-err.patch [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+Subject: [PATCH] uClibcxx:ios: fix wclog err
+
+N/A
+
+Change-Id: I5e2311c4ae906b3e3942594b98675a43f8380c3c
+---
+ src/ios.cpp | 6 +++---
+ 1 file changed, 3 insertions(+), 3 deletions(-)
+
+diff --git a/src/ios.cpp b/src/ios.cpp
```
Comment (xiaoxiang781216) [inline libs/libxx/0001-uClibcxx-ios-fix-wclog-err.patch:13]: need add prefix, please take a look other patch file in the same folder.

File: libs/libxx/0001-uclibxx-use-overload-constructor-of-filebuf-ostream.patch [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+ include/istream   |  4 +++
+ include/ostream   |  8 +++++
+ include/streambuf |  6 ++--
+ src/ios.cpp       | 84 ++++++++---------------------------------------
+ 6 files changed, 44 insertions(+), 94 deletions(-)
+
+diff --git a/uClibc++/include/fstream b/uClibc++/include/fstream
+index fddfa35..99a016e 100644
+--- a/uClibc++/include/fstream
++++ b/uClibc++/include/fstream
```
Comment (xiaoxiang781216) [inline libs/libxx/0001-uclibxx-use-overload-constructor-of-filebuf-ostream.patch:22]: Here is the prefix:
https://github.com/apache/incubator-nuttx/blob/master/libs/libxx/0001-uClibcxx-basic_definitions-fix-GCC-specific-definiti.patch#L16-L17

**Comment by yamt** [comment]
@xiaoxiang781216 why closed?

**Comment by xiaoxiang781216** [comment]
mistake, @zhuyanlinzyl will reopen again.

**Comment by zhuyanlinzyl** [comment]
https://github.com/apache/incubator-nuttx/pull/5321

Please review again.

@yamt  @xiaoxiang781216

**Pull request closed without merging**